### PR TITLE
add about, courses, credits, discord and 404 pages, plus features & fixes

### PR DESCRIPTION
**Pull request by smeijer**
**Added five pages:**

- About page
- Courses page
- Credits page
- Discord page
- 404 page

**A couple of features:**

- Mobile menu
- Hide QR code on page load
- Use an SVG in the `Load more articles` button instead of the `+` character
- Animate "back to overview link" which is found detail pages
- Animate arrow links in all directions
- Focus ring on article cards

**And a few fixes:**

- Sync spacing across pages
- Fixed a few light-mode color issues, think white-on-white text and the gradient in the chat transcript
- Fix image alignments due to new image croppings
- Reduced click area on blog filter buttons
- Center aligned blog banner images

**Other:**

The 404 page wasn't part of the design, but I created something using the standard hero component and added the "recent blogs section" below that. If wanted, we can add more content to it, like recent chats, or highlight the courses.

I've also changed the responsive breakpoints. I think the breakpoint of mobile is good like this. The medium-size breakpoint is a bit of a weird one though. That also gives the illusion that we break to mobile way too early. Medium size is basically mobile, with a few exceptions.

### CODE REVIEW

**Review by kentcdodds — COMMENTED** [review]
Super work! This wasn't all that bad to review thanks to the commits 👍 Let me know if my comments don't make any sense 😅

File: app/components/sections/hero-section.tsx [hunk 1]
```diff
@@ -1,43 +1,75 @@
...
+}
+
+function HeroSection({
+  imageUrl,
+  imageAlt,
+  imageSize = 'medium',
+  action,
+  title,
+  subtitle,
+  arrowUrl = '#',
```
[the remark this answers is not in the document]
Comment (smeijer) [reply app/components/sections/hero-section.tsx:25]: About that arrow, will it be a Link, and point to the first heading? (we'll need to add heading ids), or make it a button that scrolls down "100vh"?

[the remark this answers is not in the document]
Comment (kentcdodds) [reply app/components/sections/hero-section.tsx:25]: Let's make it a link 👍

File: app/routes/blog/index.tsx [hunk 2]
```diff
@@ -138,35 +140,18 @@ function BlogHome() {
...
-              action="/blog"
-              method="GET"
-              onSubmit={e => e.preventDefault()}
-            >
+    <>
+      <HeroSection
+        title="Learn development with great articles."
+        subtitle="Find the latest of my writing here."
+        imageUrl={images.skis()}
+        imageAlt={images.helmet.alt}
```
[the remark this answers is not in the document]
Comment (smeijer) [reply app/routes/blog/index.tsx:148]: :facepalm:

File: app/components/sections/hero-section.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (smeijer) [reply app/components/sections/hero-section.tsx]: Done. I have added some IDs and made the "arrow down" work. But I think this is also an area where we can improve.

For example, slugify header titles, so id's are auto-generated. If users could copy them as perm-link, it's also handy when sharing pages.

Maybe add progressive enhancement, and smooth-scroll downwards when the target element is close enough.

**Review by kentcdodds — APPROVED** [review]
Tons of work here. Thank you!

**Comment by kentcdodds** [comment]
Probably shouldn't have squash-merged this one 🤦 Oh well

**Pull request merged**